src/EFCore.Relational/Update/ModificationCommand.cs

```diff
@@ -419,9 +419,63 @@ private List<IColumnModification> GenerateColumnModifications()
                         columnIsCondition: false,
                         _sensitiveLoggingEnabled)));
                 }
+
+                // In TPH, the sproc has parameters for all entity types in the hierarchy; we must generate null column modifications
+                // for parameters for unrelated entity types.
+                // Enumerate over the sproc parameters in order, trying to match a corresponding parameter mapping.
+                using var parameterMappingEnumerator = storedProcedureMapping.ParameterMappings
+                    .Where(c => c.Column is IStoreStoredProcedureParameter)
+                    .OrderBy(c => ((IStoreStoredProcedureParameter)c.Column).Position)
+                    .GetEnumerator();
+                var hasParameterMapping = parameterMappingEnumerator.MoveNext();
+
+                foreach (var parameter in StoreStoredProcedure.Parameters)
+                {
+                    if (hasParameterMapping && ReferenceEquals(parameterMappingEnumerator.Current.Parameter.StoreParameter, parameter))
+                    {
+                        HandleColumnModification(parameterMappingEnumerator.Current);
+                        hasParameterMapping = parameterMappingEnumerator.MoveNext();
+                        continue;
+                    }
```

Comment:
You don't need to order the mappings
```suggestion
					var parameterMapping = storedProcedureParameter.FindParameterMapping(entry.EntityType);
                    if (parameterMapping != null)
                    {
                        HandleColumnModification(parameterMapping);
                        continue;
                    }
```

Reply:
Thanks, still learning the metadata side, apparently.